frontend/src/container/NewWidget/LeftContainer/QuerySection/QueryBuilder/clickHouse/index.tsx

```diff
@@ -42,7 +42,7 @@ function ClickHouseQueryContainer({
 			currentIndexQuery.disabled = !currentIndexQuery.disabled;
 		}
 		if (toggleDelete) {
-			allQueries.splice(queryIndex, 1);
+			allQueries.splice(queryIndex as number, 1);
```

Comment:
is it possible to make a type check of number

Reply:
not sure if that will help. this function should be called only when queryIndex is number. but the way the metric builder components are written, I had to resort this workaround. 

the problem is metric builder for some reason, changes the queries from 
[A]: query to [1]: {query}. This was probably due to change in the backend at that time from [1]: query to [A]:query format 

But in alert builder, we represent the query same as backend [A]: query. 

Please feel free to suggest an alternative here.

Reply:
thanks

Reply:
done